### PR TITLE
added search functionality to OnlineDoctorsSelect

### PR DESCRIPTION
**Pull request by khavinshankar**
this resolves issue #1269 

image

### CODE REVIEW

**Comment by bodhish** [comment]
@khavinshankar the current version will only search on the 1st 14 entries got on the 1st API request. 

It should ideally be a server-connected search. The search response should come from the server.

**Comment by samholics** [comment]
testd, verified

it shows the search bar nd getting the reults

File: src/Components/Common/OnlineDoctorsSelect.tsx [hunk 1]
```diff
@@ -38,8 +59,17 @@ export const OnlineDoctorsSelect = (props: any) => {
...
           <span className="inline-block w-full rounded-md shadow-sm">
-            <button onClick={_ => setIsExpanded(!isExpanded)} type="button" aria-haspopup="listbox" aria-expanded="true" aria-labelledby="listbox-label" className="cursor-default relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue-300 transition ease-in-out duration-150 sm:text-sm sm:leading-5">
-              <div className="flex items-center justify-between">
+            <button onClick={_ => setIsExpanded(!isExpanded)} type="button" aria-haspopup="listbox" aria-expanded="true" aria-labelledby="listbox-label" className={`cursor-default relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left focus:outline-none focus:shadow-outline-blue focus:border-blue-300 transition ease-in-out duration-150 sm:text-sm sm:leading-5`}>
+              <input
+                ref={searchFieldRef}
+                name="searchTerm"
+                type="text"
+                placeholder="Search by username"
+                className={`py-2 pl-3 w-full outline-none ${!isExpanded ? 'hidden' : ''}`}
```
Comment (smartclash) [inline src/Components/Common/OnlineDoctorsSelect.tsx]: Can we use `classnames` library here? It's already a dependency

File: src/Components/Common/OnlineDoctorsSelect.tsx [hunk 2]
```diff
@@ -38,8 +59,17 @@ export const OnlineDoctorsSelect = (props: any) => {
...
+                value={state.searchTerm}
+                onChange={handleSearchTermChange}
+              />
+              <div className={`flex items-center justify-between ${isExpanded ? 'hidden' : ''}`}>
```
Comment (smartclash) [inline src/Components/Common/OnlineDoctorsSelect.tsx]: ... and here too.

File: src/Components/Common/OnlineDoctorsSelect.tsx [hunk 3]
```diff
@@ -64,7 +94,7 @@ export const OnlineDoctorsSelect = (props: any) => {
             <div role="listbox" aria-labelledby="listbox-label" aria-activedescendant="listbox-item-3" className="multiselect-dropdown__search-dropdown w-full absolute border border-gray-400 bg-white mt-1 rounded-lg shadow-lg px-4 py-2 z-50">
               {
                 state.users.map((user: any) => {
-                  return <button onClick={_ => { setIsExpanded(false); onSelect(user.id) }} id="listbox-item-0" role="option" className="flex text-xs py-1 items-center w-full hover:bg-gray-200 focus:outline-none focus:bg-gray-200">
+                  return <button onClick={_ => { setIsExpanded(false); onSelect(user.id); setState({...state, searchTerm: ""}) }} id="listbox-item-0" role="option" className="flex text-xs py-1 items-center w-full hover:bg-gray-200 focus:outline-none focus:bg-gray-200">
```
Comment (smartclash) [inline src/Components/Common/OnlineDoctorsSelect.tsx]: Also, add an `id` attribute

**Review by smartclash — CHANGES_REQUESTED** [review]

File: src/Components/Common/OnlineDoctorsSelect.tsx (hunk 3 above)
Comment (bodhish) [reply src/Components/Common/OnlineDoctorsSelect.tsx]: Add a `key` too

File: src/Components/Common/OnlineDoctorsSelect.tsx (hunk 1 above)
Comment (khavinshankar) [reply src/Components/Common/OnlineDoctorsSelect.tsx]: nice 😃

File: src/Components/Common/OnlineDoctorsSelect.tsx (hunk 2 above)
Comment (khavinshankar) [reply src/Components/Common/OnlineDoctorsSelect.tsx]: 👍

File: src/Components/Common/OnlineDoctorsSelect.tsx (hunk 3 above)
Comment (khavinshankar) [reply src/Components/Common/OnlineDoctorsSelect.tsx]: 👍

**Review by smartclash — APPROVED** [review]
LGTM :+1:

**Pull request merged**